src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirOperationDataStore.cs

```diff
@@ -129,16 +157,32 @@ public async Task<ExportJobOutcome> UpdateExportJobAsync(ExportJobRecord jobReco
             using (SqlConnectionWrapper sqlConnectionWrapper = await _sqlConnectionWrapperFactory.ObtainSqlConnectionWrapperAsync(cancellationToken, true))
             using (SqlCommandWrapper sqlCommandWrapper = sqlConnectionWrapper.CreateSqlCommand())
             {
+                async void OnRetry(Exception exception, TimeSpan timeSpan, int i, Context context)
```

Comment:
We may be able to refactor this such that the `OnRetry` object is re-used across all methods. I think the only thing we'd need to do is probably read the `Parameters` from the `Context` object as opposed to capturing it in the method as `sqlCommandWrapper.Parameters`

Reply:
If we do that can have the resulting `AsyncRetryPolicy` as the class member as opposed to the policy builder

Reply:
As discussed, it makes sense to avoid setting up the `AsyncRetryPolicy` in the `OnRetry` Actions. Since the `OnRetry` Actions are operation-specific and can't be combined, we could create two custom policies that are set up in the constructor. This would prevent us from having to create policies at runtime.

Additionally, we could avoid using free variables in the `OnRetry` Actions to avoid the work that would need to be done in the background to capture the current state before `OnRetry` is executed (I used [this](https://stackoverflow.com/questions/595482/what-are-closures-in-c) to review a brief overview of closures).